### PR TITLE
feat(dflash): clarify W&B metrics and diagnose prefix acceptance failures

### PR DESCRIPTION
**Pull request by maocheng23**
## Motivation

DFlash2's `serving_accuracy` scores every supervised position, including tokens after an earlier mismatch. Its `conditional_accuracy` conditions on candidate coverage, rather than a correct preceding prefix. These names make it difficult to tell whether short accepted lengths come from early errors, later errors, missing candidates, or selector ranking.

This PR makes those populations explicit and adds prefix diagnostics for unary greedy, top-K oracle, and selector greedy paths on recorded training blocks.

## Modifications

- Rename the marginal and covered selector metrics, and label gold-probability / teacher-overlap chain lengths as training proxies. Remove duplicate `expected_acceptance`, `selector_loss`, and `selector_coverage` aliases; clarify the weighted selector objective metrics.
- Add per-position prefix acceptance, survival, reached/accepted/first-failure counts, and unary greedy accepted length. On selector-reached prefixes, report unary accuracy, top-K recall, covered selector accuracy, and the split between candidate misses and ranking errors.
- Report supervised-prefix counts and length so masked tails are distinguishable from model failures. Preserve the trainer's zero-denominator convention and document how counts identify unobserved rates.
- Carry additive counts through gradient accumulation and the existing packed DP sum before calculating rates. Reuse the existing diagnostic logits and greedy path, with no additional LM-head call or distributed collective.
- Add `train/optimizer_step` as an explicit checkpoint/eval join key and document every dashboard rename. Existing W&B history is unchanged.

These metrics describe fixed training blocks; they do not claim stochastic serving acceptance or a cross-block serving walk. Objectives, model parameters, and checkpoint formats are unchanged.

## Related Issues

Follow-up to #810.

## Accuracy Test

135 CPU tests and 12 subtests passed with Python 3.12, PyTorch 2.13.0, and Transformers 5.12.1:

```bash
python -m pytest -q \
  tests/test_utils/test_dflash_metrics.py \
  tests/test_utils/test_dflash_losses.py \
  tests/test_runtime/test_trainer.py \
  tests/test_runtime/test_tracking_logger.py \
  tests/test_runtime/test_evaluator_aggregation.py \
  tests/test_runtime/test_seam_fixes.py \
  tests/test_modeling/test_dflash2.py
```

Coverage includes hand-counted early/late failures, candidate-versus-ranking errors, masked tails and internal gaps, empty prefixes, plain DFlash, chunk invariance, unequal DP populations, accumulation/reset behavior, strategy-to-tracker plumbing, loss/gradient parity with diagnostics enabled or disabled, and the existing two-process Gloo eval regression.

`pre-commit run --all-files` and `git diff --check` passed.

## Benchmark & Profiling

No GPU training-throughput benchmark or serving eval was run. Detailed diagnostics retain the existing log-step gate; the added work consists of prefix masks, small reductions, and scalar telemetry. This draft is ready for code review and a GPU smoke test before rollout.

## Checklist

- [x] Run the repository formatting hooks.
- [x] Add and run relevant unit tests.
- [x] Update documentation, including the dashboard migration table.
- [ ] GPU smoke test / throughput measurement before rollout.

### CODE REVIEW

**Review by jiapingW — COMMENTED** [review]
LGTM.
